modules/caddyhttp/server.go

```diff
@@ -394,6 +355,57 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *Server) logRequest(
+	accLog *zap.Logger, r *http.Request, wrec ResponseRecorder, duration *time.Duration,
+	repl *caddy.Replacer, bodyReader *lengthReader, shouldLogCredentials bool,
+) {
+	// this request may be flagged as omitted from the logs
+	if skipLog, ok := GetVar(r.Context(), SkipLogVar).(bool); ok && skipLog {
+		return
+	}
+
+	repl.Set("http.response.status", wrec.Status()) // will be 0 if no response is written by us (Go will write 200 to client)
+	repl.Set("http.response.size", wrec.Size())
+	repl.Set("http.response.duration", duration)
+	repl.Set("http.response.duration_ms", duration.Seconds()*1e3) // multiply seconds to preserve decimal (see #4666)
+
+	logger := accLog
+	if s.Logs != nil {
+		logger = s.Logs.wrapLogger(logger, r.Host)
+	}
+
+	log := logger.Info
+	if wrec.Status() >= 400 {
+		log = logger.Error
+	}
+
+	userID, _ := repl.GetString("http.auth.user.id")
+
+	reqBodyLength := 0
+	if bodyReader != nil {
+		reqBodyLength = bodyReader.Length
+	}
+
+	// preallocate fields with a capacity of 7 in case traceID is set
+	fields := make([]zapcore.Field, 0, 7)
+	fields = append(fields, zap.Int("bytes_read", reqBodyLength))
+	fields = append(fields, zap.String("user_id", userID))
+	fields = append(fields, zap.Duration("duration", *duration))
+	fields = append(fields, zap.Int("size", wrec.Size()))
+	fields = append(fields, zap.Int("status", wrec.Status()))
+	fields = append(fields, zap.Object("resp_headers", LoggableHTTPHeader{
+		Header:               wrec.Header(),
+		ShouldLogCredentials: shouldLogCredentials,
+	}))
+
+	traceID, _ := GetVar(r.Context(), "traceID").(string)
+	if traceID != "" {
+		fields = append(fields, zap.String("traceID", traceID))
```

Comment:
All our log fields are snake_case, should be consistent

```suggestion
		fields = append(fields, zap.String("trace_id", traceID))
```

Reply:
IMO `traceID` is best as that's the default used by a number of tools. If we use snake-case for this we can modify tools, but that just makes it more difficult to do log/trace correlation with Caddy.